Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -180,7 +180,16 @@ namespace Babylon::Polyfills::Internal
     {
         try
         {
-            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+            auto inputURL{info[1].As<Napi::String>()};
+            auto decodedURL{info.Env().Global().Get("decodeURI").As<Napi::Function>().Call({inputURL}).As<Napi::String>()};
+            auto encodedURL{inputURL};
+            // We need to percent encode the URL
+            // If decoding the URL changes something, it's already encoded - don't change anything
+            if (inputURL.StrictEquals(decodedURL))
+            {
+                encodedURL = info.Env().Global().Get("encodeURI").As<Napi::Function>().Call({inputURL}).As<Napi::String>();
+            }
```

Comment:
I'm not sure I follow this logic. What happens if the URL has both encoded spaces and not encoded spaces?

Reply:
If we want to handle that case (a partially-encoded URL), we will need to write our own custom encoding function. We can do this, but then we are responsible for handling all edge cases. encodeURI is standardized.

I'm not sure how likely it is that we would get a string like this. I'd be surprised if someone encoded only part of their URL, but I guess it is possible. (Maybe they concatenate "some%20encoded%20url" + "some non-encoded string".)

If we need to support this case, I can write a function in XMLHttpRequest to do this. At least it would be cross-platform that way.